src/Http/User/UserIdentityController.php

```diff
@@ -23,41 +24,53 @@ class UserIdentityController
     private $userIdentityResolver;
 
     /**
-     * @var string
+     * @var JsonWebToken
      */
-    private $currentUserId;
+    private $jwt;
 
     public function __construct(
         UserIdentityResolver $userIdentityResolver,
-        string $currentUserId
+        JsonWebToken $jsonWebToken
     ) {
         $this->userIdentityResolver = $userIdentityResolver;
-        $this->currentUserId = $currentUserId;
+        $this->jwt = $jsonWebToken;
     }
 
     public function getByEmailAddress(ServerRequestInterface $request): JsonResponse
     {
+        $notFoundResponse = new ApiProblemJsonResponse(
+            ApiProblems::userNotFound('No user found for the given email address.')
+        );
+
         try {
             $emailAddress = new EmailAddress($request->getAttribute('emailAddress'));
         } catch (InvalidNativeArgumentException $e) {
-            return $this->createUserNotFoundResponse();
+            return $notFoundResponse;
```

Comment:
Is this not a bad request instead of a not found request?

Reply:
I'm not sure what you mean? It uses status code `404` like it did before. And the path is `users/emails/{emailAddress}`, for example `users/emails/luc@madewithlove.be` if the email address is not found that endpoint does not exist so a `404` no?

Reply:
Indeed the actual status code is not changed.
But with a fresh look on the errors I wonder what it means if the email address is not present inside the request or has a wrong format. Is that not a bad request?

Reply:
Oh yes I see now what you mean and you are 100% correct, I'll change it!

Reply:
I thought this was about the user not being found when searching by email

Reply:
I also forgot to change the "user not found" back to a `404` as it should be for the search by email address (if the email is valid), so I also did that